apps/TutorialServo/TutorialServo.cpp

```diff
@@ -0,0 +1,44 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "TutorialServo.h"
+#include "mbed.h"
+#include <cstdlib>
+
+
+void TutorialServo::setAngleRangeInDegrees(float degrees)
+{
+  angleRange = degrees;
+}
+
+void TutorialServo::setPulseWidthRangeInMs(float minPulseWidthMsFloat, float maxPulseWidthMsFloat)
+{
+  minPulseWidthMs = minPulseWidthMsFloat;
+  maxPulseWidthMs = maxPulseWidthMsFloat;
+}
+
+void TutorialServo::setPositionInDegrees(float degrees)
+{
+  if(!((std::fabs(degrees) > angleRange)))
+    {
+        double pulseWidthMs = 1.5 + 0.5*degrees/angleRange; //Sets the pulse width that will be used, as 1ms is the lower bound and 2ms is the upper bound.
+        //1.5 is neutral and degrees can be either positive or negative which can make a pulse with from 1-2ms
+        if((minPulseWidthMs < pulseWidthMs) && (pulseWidthMs < maxPulseWidthMs)) //checks to ensure it is within bounds
+        {
+           servoPwm.pulsewidth(pulseWidthMs/1000); //set pulse width
+        }     
+        
+    }
+}
+
+TutorialServo::TutorialServo(PinName servoPin, float servoFrequency) : servoPwm(servoPin) //Constructor
```

Comment:
users should have the choice whether to specifiy things like min and max pulse widths and angleRange. Now, if they choose to not specify these, the defaults should be used.

Reply:
They way it is written now, defaults are always used.

Reply:
Okay, would it be correct to then overload the function with multiple constructors that slowly go up?

Reply:
you could do this, but melvin wang shared a video in the slack you can check out abt core guidelines which discusses this precisely. Essentially, it is better to have default params that overloaded constructors.